src/ompl/geometric/planners/bitstar/src/BITstar.cpp

```diff
@@ -914,11 +909,32 @@ namespace ompl
             //A new start pointer
             const ompl::base::State* newStart;
 
+            //If the list of goal vertices is empty, wait until the first goal state is available.
+            if (goalVertices_.empty())
+            {
+                newGoal = Planner::pis_.nextGoal(ptc);
+                if (newGoal)
+                {
+                    //It is valid, allocate the vertex pointer
+                    goalVertices_.push_back(boost::make_shared<Vertex>(Planner::si_, opt_));
+
+                    //Copy the value into the state
+                    Planner::si_->copyState(goalVertices_.back()->state(), newGoal);
+
+                    //And add this goal to the set of samples:
+                    this->addSample(goalVertices_.back());
+
+                    //Mark that we've added:
+                    addedState = true;
+                }
+            }
+
             //Add the new starts and goals to the lists of said vertices.
             //Do goals first, as they are only added as samples.
-            //If newGoal is NULL, then no new goal exists. The following line both assigns newGoal and checks if it is a valid pointer.
-            while (static_cast<bool>((newGoal = Planner::pis_.nextGoal(ptc))) == true)
+            while (pis_.haveMoreGoalStates())
```

Comment:
Can you explain why you made this change, was it not working? nextGoal should return NULL if there isn't another goal.

While I **hate** the condition for the while loop (it is probably the most obfuscated line of code I've ever written), it seemed preferable to needing the code duplication for the two goal cases...